Makefile

```diff
@@ -323,6 +325,9 @@ $(info --------------)
 $(info )
 $(info )
 
+ifneq (, $(TARGET))
+	TARGET := $(TARGET)/
+endif
```

Comment:
How about TARGET_DIR := target/$(TARGET)/release when true or TARGET_DIR := target/release when TARGET is blank? Then we can simplify a lot of paths.

My underlying problem with this is that a single variable with one name switched its "logical type" from being a string containing just the target type, to being a directory name.